target_chains/ethereum/sdk/solidity/PythUtils.sol

```diff
@@ -33,4 +33,38 @@ library PythUtils {
                 10 ** uint32(priceDecimals - targetDecimals);
         }
     }
+
+    /// @notice Combines two prices to get a cross-rate
+    /// @param price1 The first price (a/b)
+    /// @param price2 The second price (c/b)
+    /// @return crossRate The cross-rate (a/c)
+    /// @return expo The exponent of the cross-rate
+    /// @dev This function will revert if either price is negative or if the exponents are invalid.
+    /// @dev This function will also revert if the cross-rate is greater than int64.max
+    /// @notice This function doesn't return the combined confidence interval.
+    function deriveCrossRate(
+        int64 price1,
+        int32 expo1,
+        int64 price2,
+        int32 expo2
+    ) public pure returns (int64 crossRate, int32 expo) {
+        if (price1 < 0 || price2 < 0 || expo1 > 0 || expo2 > 0 || expo1 < -255 || expo2 < -255) {
+            revert();
+        }
+
+        // Convert both prices to the same decimal places (using the larger of the two)
+        uint8 maxDecimals = uint8(uint32(-1 * (expo1 < expo2 ? expo1 : expo2)));
+        uint256 p1 = convertToUint(price1, expo1, maxDecimals);
+        uint256 p2 = convertToUint(price2, expo2, maxDecimals);
+
+        // Calculate the combined price with precision
+        uint256 combined = (p1 * 10**18) / p2;
+        combined = combined / 10 ** (18 - maxDecimals);
```

Comment:
the other case to be very careful about is when this method can fail due to numeric overflow. This is another gotcha that could trigger if the price of asset goes up a lot.

(I think you're pretty unlikely to overflow given the real-world range of the exponents, but good to be explicit about that. You may even want to restrict the exponent range in the assert further to guarantee that there's no overlfow)

Reply:
sorry this is in reference to the multiplication on line 61